### PR TITLE
Update support for langfuse metadata

### PR DESCRIPTION
**Pull request by alexanderepstein**
## Title

Update support for langfuse metadata

## Relevant issues

Builds upon #3221 

## Type

🆕 New Feature
🧹 Refactoring
📖 Documentation
✅ Test

## Changes

- Added ability to set trace release, version, metadata
- Added ability to update fields during a trace continuation
- Added ability to update input and output during a trace continuation
- Wrote new test for verifying metadata is set correctly
- Small improvement to setting secret boolean, prevent unnecessary literal_eval
- Small improvements to langfuse tests

## Testing

- Ran existing test suite along with new test

## Notes

- Updates to support most langfuse metadata facilities

## Pre-Submission Checklist (optional but appreciated):

- [x] I have included relevant documentation updates (stored in /docs/my-website)

## OS Tests (optional but appreciated):

- [ ] Tested on Windows
- [ ] Tested on MacOS
- [x] Tested on Linux

### CODE REVIEW

File: litellm/tests/test_alangfuse.py [hunk 1]
```diff
@@ -211,6 +214,75 @@ async def test_langfuse_logging_without_request_response(stream):
         pytest.fail(f"An exception occurred - {e}")
 
 
+@pytest.mark.asyncio
+async def test_langfuse_logging_metadata(langfuse_client):
```
Comment (ghost) [inline litellm/tests/test_alangfuse.py:225]: hey @alexanderepstein can you add some comments on this test - would help understand what it tests + expected behaviour

File: litellm/tests/test_alangfuse.py [hunk 2]
```diff
@@ -161,38 +164,38 @@ async def make_async_calls():
...
+            "model": "azure/chatgpt-v-2",
+            "messages": [{"role": "user", "content": "This is a test"}],
+            "max_tokens": 5,
+            "temperature": 0.7,
+            "timeout": 5,
+            "user": "langfuse_latency_test_user",
+            "mock_response": "It's simple to use and easy to get started",
+        }
+    completion_args.update(completion_kwargs)
+    return asyncio.create_task(litellm.acompletion(**completion_args))
```
Comment (ghost) [inline litellm/tests/test_alangfuse.py:185]: is the create task being run outside the function? i can see 'create_Async_task' being called inside some tests - what's happening here?

File: docs/my-website/docs/observability/langfuse_integration.md [hunk 3]
```diff
@@ -121,12 +124,21 @@ response = completion(
...
-      "tags": ["tag1", "tag2"]                      # set langfuse Tags
+      "tags": ["tag1", "tag2"],                     # set langfuse Tags
       "trace_id": "trace-id22",                     # set langfuse Trace ID
+      "trace_metadata": {"key": "value"},           # set langfuse Trace Metadata
+      "trace_version": "test-trace-version",        # set langfuse Trace Version (if not set, defaults to Generation Version)
+      "trace_release": "test-trace-release",        # set langfuse Trace Release
       ### OR ### 
-      "existing_trace_id": "trace-id22",                     # if generation is continuation of past trace. This prevents default behaviour of setting a trace name
+      "existing_trace_id": "trace-id22",            # if generation is continuation of past trace. This prevents default behaviour of setting a trace name
+      ### OR enforce that fields are trace overwritten in the trace during the continuation ###
```
Comment (ghost) [inline docs/my-website/docs/observability/langfuse_integration.md]: I think? this means allow overwriting specific trace params 

Which params 'can' be overwritten and which can't? 

A suggestion - if you were reading our langfuse docs for the first time, how could this be simply explained for you?

**Comment by ghost** [comment]
@alexanderepstein added some comments. 

If we can address those + attach a screenshot of the langfuse testing passing for you - i think this should be good to merge

**Comment by alexanderepstein** [comment]
I am not sure I see your comments, let me know if I'm missing something.

Had to add mark one test as skippable if you don't have an OpenAI api key.

Here is the test output for the file.

litellm-langfuse-tests

**Comment by ghost** [comment]
Screenshot 2024-05-04 at 11 00 54 PM

are you not able to see these? @alexanderepstein

**Comment by alexanderepstein** [comment]
Looks like your comments are pending https://github.com/orgs/community/discussions/10369

**Comment by alexanderepstein** [comment]
I attempted to address the comments I can see in your screenshot.

**Pull request merged**